app/options.go

```diff
@@ -127,9 +125,13 @@ func StartSearchEngine(s *Server) error {
 	return nil
 }
 
+// SetLogger requires platform service to be initialized before calling.
+// If not, logger should be set after platform service are initialized.
 func SetLogger(logger *mlog.Logger) Option {
 	return func(s *Server) error {
-		s.Log = logger
+		if s.platform != nil {
+			s.platform.SetLogger(logger)
+		}
 		return nil
```

Comment:
Is it a good idea to fail silently here, when `s.platform == nil`? It seems to me like that should at least be logged.

Reply:
This is being used for the testing purposes, I don't think logging is necessary here.